core/src/main/java/hudson/model/labels/LabelAtom.java

```diff
@@ -195,12 +197,14 @@ public void save() throws IOException {
 
     public void load() {
         XmlFile file = getConfigFile();
-        if(file.exists()) {
-            try {
+        try {
+            if (file.exists()) {
                 file.unmarshal(this);
-            } catch (IOException e) {
-                LOGGER.log(Level.WARNING, "Failed to load "+file, e);
             }
+        } catch (InvalidPathException e) {
+            LOGGER.log(Level.FINE, "Failed to load " + file, e);
+        } catch (IOException | UncheckedIOException e) {
+            LOGGER.log(Level.WARNING, "Failed to load " + file, e);
         }
```

Comment:
Needed to avoid errors like this on Windows:

```java
java.nio.file.InvalidPathException: Illegal char <<> at index 84: C:\Jenkins\workspace\Core_jenkins_PR-6036\test\target\j h1212842842721259356\labels\<img\src='x' onerror=alert(123)>xss.xml
	at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
	at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:229)
	at java.base/java.io.File.toPath(File.java:2329)
	at hudson.XmlFile.exists(XmlFile.java:247)
	at hudson.model.labels.LabelAtom.load(LabelAtom.java:198)
	at jenkins.model.Jenkins.getLabelAtom(Jenkins.java:2088)
	at jenkins.model.Jenkins.getLabel(Jenkins.java:2066)
	at hudson.model.Label.get(Label.java:601)
```

Since the `InvalidPathException` is expected in many cases, we catch it and log it at `Level.FINE` to avoid creating unnecessary log spam.

Reply:
I thought I remembered some utility to abstract this, but all I see is https://github.com/jenkinsci/jenkins/blob/f29b4789834a7562b44d8c3a0abcd9fb0c927f27/core/src/main/java/hudson/Util.java#L1717-L1728

Reply:
Thanks for pointing this out. I used `Util.fileToPath` anywhere that a method was declared to throw `IOException`.